tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -836,17 +835,126 @@ export class WebGPUBackend extends KernelBackend {
     return output;
   }
 
+  private getFromPixelTextureLayout(useImport: boolean): WebGPULayout {
+    if (useImport) {
+      if (this.fromPixelImportTextureLayout === null) {
+        this.fromPixelImportTextureLayout =
+            this.createFromPixelTextureLayout(true /* useImport */);
+      }
+      return this.fromPixelImportTextureLayout;
+    }
+
+    if (this.fromPixelTextureLayout === null) {
+      this.fromPixelTextureLayout =
+          this.createFromPixelTextureLayout(false /* useImport */);
+    }
+    return this.fromPixelTextureLayout;
+  }
+
+  private createFromPixelTextureLayout(useImport: boolean): WebGPULayout {
+    const bindGroupLayoutEntries: GPUBindGroupLayoutEntry[] = [];
+    // Output buffer binding layout.
+    bindGroupLayoutEntries.push({
+      binding: 0,
+      visibility: GPUShaderStage.COMPUTE,
+      buffer: {type: 'storage' as const}
+    });
+    // Input texture binding layout.
+    if (useImport) {
+      bindGroupLayoutEntries.push({
+        binding: 1,
+        visibility: GPUShaderStage.COMPUTE,
+        externalTexture: {},
+      });
+    } else {
+      bindGroupLayoutEntries.push(
+          {binding: 1, visibility: GPUShaderStage.COMPUTE, texture: {}});
+    }
+    // Uniform buffer binding layout.
+    bindGroupLayoutEntries.push(
+        {binding: 2, visibility: GPUShaderStage.COMPUTE, buffer: {}});
+    const fromPixelBindGroupLayout =
+        this.device.createBindGroupLayout({entries: bindGroupLayoutEntries});
+    const fromPixelPipelineLayout = this.device.createPipelineLayout(
+        {bindGroupLayouts: [fromPixelBindGroupLayout]});
+    return {
+      bindGroupLayout: fromPixelBindGroupLayout,
+      pipelineLayout: fromPixelPipelineLayout
+    };
+  }
+
+  private copyExternalImageToTexture(
+      externalImage: ExternalImage|HTMLVideoElement,
```

Comment:
nit: `ExternalImage|HTMLVideoElement` -> `ExternalImage`? I think `HTMLVideoElement` is not supported by `copyExternalImageToTexture`.

Reply:
thanks, done